packages/plugin-ext/src/main/browser/view/plugin-view-registry.ts

```diff
@@ -780,19 +780,48 @@ export class PluginViewRegistry implements FrontendApplicationContribution {
         return toDispose;
     }
 
+    checkViewDataProvider(): void {
+        for (const viewId of this.views.keys()) {
+            if (!this.viewDataProviders.has(viewId)) {
+                this.getView(viewId).then(async view => {
+                    if (view) {
+                        if (view.isVisible) {
+                            await this.prepareView(view);
+                        } else {
+                            const toDisposeOnDidExpandView = new DisposableCollection(this.onDidExpandView(async id => {
+                                if (id === viewId) {
+                                    unsubscribe();
+                                    await this.prepareView(view);
+                                }
+                            }));
+                            const unsubscribe = () => toDisposeOnDidExpandView.dispose();
+                            view.disposed.connect(unsubscribe);
+                            toDisposeOnDidExpandView.push(Disposable.create(() => view.disposed.disconnect(unsubscribe)));
+                        }
+                    }
+                });
+            }
+        }
+    }
```

Comment:
This code duplicates a lot of what `registerViewDataProvider` already does, I don't think it is the proper solution as is.

Reply:
May I ask if the meaning here is that I should try to reuse code? Or should we solve the problem in a different way

Reply:
Sorry, I mean that we should not have duplication, and it looks like the current solution duplicates a lot of what `registerViewDataProvider` already does without any clear reason as to why it does. We should find a solution which does not duplicate the code, either by seeing what can be reused once we better understand the problem, or a completely different solution if necessary.

Reply:
Okay, I understand. In Theia，widgets should be opened only when view data providers are registered, but some vscode plugins only have some welcomeViews and do not register a viewDataProvider. So I want to check if the plugin view where the welcome view is located has been registered a viewDataProvider after the plugin is launched, and if not, give a default

Reply:
Thank you for your previous suggestion, but I can't find a simpler solution. I will create a new PR to remove the commit messages and wait for other results